packages/trader/src/Modules/Trading/Components/Elements/Multiplier/__tests__/cancel-deal-mobile.spec.tsx

```diff
@@ -0,0 +1,143 @@
+        const new_mock_store = { ...default_mock_store, ui: { should_show_cancellation_warning: true } };
+        const mock_root_store = mockStore(new_mock_store);
+        render(mockCancelDeal(mock_root_store, new_mocked_props));
+
+        expect(screen.queryByText(dialog_text)).not.toBeInTheDocument();
+        expect(screen.queryByText(check_box_text)).not.toBeInTheDocument();
+
+        const checkbox_with_cancellation = screen.getByText(deal_cancellation);
+        userEvent.click(checkbox_with_cancellation);
+
+        expect(screen.getByText(dialog_text)).toBeInTheDocument();
+        expect(screen.getByText(check_box_text)).toBeInTheDocument();
+    });
+    it('should call toggleCancellationWarning function if cancellation warning checkbox inside of <DealCancellationWarningDialog /> was clicked', () => {
+        const new_mocked_props = { ...default_mocked_props, has_take_profit: true };
+        const new_mock_store = {
+            ...default_mock_store,
+            ui: { should_show_cancellation_warning: true, toggleCancellationWarning: jest.fn() },
+        };
+        const mock_root_store = mockStore(new_mock_store);
+        render(mockCancelDeal(mock_root_store, new_mocked_props));
+
+        const checkbox_with_cancellation = screen.getByText(deal_cancellation);
+        userEvent.click(checkbox_with_cancellation);
+        const checkbox_with_show = screen.getByText(check_box_text);
+        userEvent.click(checkbox_with_show);
+
+        expect(new_mock_store.ui.toggleCancellationWarning).toBeCalled();
+    });
+    it('<DealCancellationWarningDialog /> should not be visible if Cancel button was clicked', () => {
+        const new_mocked_props = { ...default_mocked_props, has_stop_loss: true };
+        const new_mock_store = { ...default_mock_store, ui: { should_show_cancellation_warning: true } };
+        const mock_root_store = mockStore(new_mock_store);
+        render(mockCancelDeal(mock_root_store, new_mocked_props));
+
+        const checkbox_with_cancellation = screen.getByText(deal_cancellation);
+        userEvent.click(checkbox_with_cancellation);
+        const cancel_button = screen.getByText('Cancel button');
+        userEvent.click(cancel_button);
+
+        expect(screen.queryByText(dialog_text)).not.toBeInTheDocument();
+        expect(screen.queryByText(check_box_text)).not.toBeInTheDocument();
+    });
+    it('<DealCancellationWarningDialog /> should not be visible if Confirm button was clicked', () => {
+        const new_mocked_props = { ...default_mocked_props, has_stop_loss: true };
+        const new_mock_store = { ...default_mock_store, ui: { should_show_cancellation_warning: true } };
+        const mock_root_store = mockStore(new_mock_store);
+        render(mockCancelDeal(mock_root_store, new_mocked_props));
+
+        const checkbox_with_cancellation = screen.getByText(deal_cancellation);
+        userEvent.click(checkbox_with_cancellation);
```

Comment:
ditto )

Reply:
Please, refer to https://github.com/binary-com/deriv-app/pull/9586/files#r1398838863